packages/nimble-components/build/generate-workers/source/renderWorker.ts

```diff
@@ -0,0 +1,167 @@
+            colIndex: new Int32Array(),
+            rowIndex: new Int32Array(),
+            value: new Float32Array()
+        });
+    }
+
+    public rerenderDies(): void {
+        for (const tableRow of this.dieTable) {
+            const scaledX = this.horizontalScale.a + this.horizontalScale.b * tableRow.colIndex + this.margin.right;
+            if (
+                scaledX >= this.xLimits.min
+                && scaledX < this.xLimits.max
+            ) {
+                const scaledY = this.verticalScale.a + this.verticalScale.b * tableRow.rowIndex + this.margin.top;
+                if (
+                    scaledY >= this.yLimits.min
+                    && scaledY < this.yLimits.max
+                ) {
+                    const category = this.colorCategories.find(cat => cat.startValue <= tableRow.value && (cat.endValue === undefined || cat.endValue > tableRow.value));
+                    this.context.fillStyle = category ? category.color : 'blue';
+                    this.context.fillRect(scaledX, scaledY, this.dieDimensions.width, this.dieDimensions.height);
+                }
+            }
+        }
+    }
+
+    public rerenderText(): void {
+        const dieSize = this.dieDimensions.width * this.dieDimensions.height * (this.transform.k || 1);
+        if (dieSize >= 50) {
+            const approximateTextHeight = this.context.measureText('M').width;
+            this.context.font = `${Math.min(this.dieDimensions.height, this.dieDimensions.width * 0.8).toString()}px sans-serif`;
+            this.context.fillStyle = '#ffffff';
+            this.context.textAlign = 'center';
+            this.context.lineCap = 'butt';
+
+            
+        for (const tableRow of this.dieTable) {
+                const scaledX = this.horizontalScale.a + this.horizontalScale.b * tableRow.colIndex + this.margin.right;
+                if (
+                    scaledX >= this.xLimits.min
+                    && scaledX < this.xLimits.max
+                ) {
+                    const scaledY = this.verticalScale.a + this.verticalScale.b * tableRow.rowIndex + this.margin.top;
+                    if (
+                        scaledY >= this.yLimits.min
+                        && scaledY < this.yLimits.max
+                    ) {
+                        this.context.fillText(
+                            tableRow.value.toFixed(1),
+                            scaledX + this.dieDimensions.width / 2,
+                            scaledY + this.dieDimensions.height - approximateTextHeight,
+                            this.dieDimensions.width - (this.dieDimensions.width / 5)
+                        );
+                    }
+                }
+            }
+        }
+    }
+
+    public renderDies(
+            tableBuffer: Uint8Array
+    ): void {
+        const start = this.performanceTest !== undefined ? self.performance.now() : undefined;
+
+       this.dieTable = tableFromIPC(tableBuffer);
```

Comment:
@rajsite the metod of transferring the IPC buffers is [suggested here](https://uwdata.github.io/arquero/api/table#toArrowBuffer) too. this transfer seems to have an impact on the performance so I will try the [geoarrow method](https://github.com/geoarrow/geoarrow-js?tab=readme-ov-file#web-worker-utilities) and compare.

Reply:
I think today we would want to have the user create an Arrow Table and transfer it to the worker like the geoarrow method.
I could see it being useful in the future to allow users to assign Arrow Table instances or ArrowIPC instances, for example if the backend started returning ArrowIPC we could transfer that directly to the worker. But I think today serializing as ArrowIPC first just adds overhead.

Reply:
I got the ideea from geoarrow to split the table in buffers which can get transferred, but there is no need to re-create the table itself in the worker